scripts/transformations.js

```diff
@@ -90,23 +90,20 @@ function toggleSpaces(pre) {
 
 /* Do not consider signature marker as a deleted line */
 function patchGitSendSignature(node) {
-    if (node.className != "moz-txt-sig")
-        return;
-    if (node.childNodes.length != 2)
+    let lastCodeNode = node.lastChild;
+
+    if (lastCodeNode.childNodes.length < 2)
         return;
 
-    let marker = node.childNodes[0];
-    let version = node.childNodes[1];
-    let versionPattern = /^\n\d+\.\d+\.\S+\n+$/;
+    let marker = lastCodeNode.childNodes[lastCodeNode.childNodes.length - 2];
 
-    if (marker.textContent != "-- ")
+    if (marker.className !== "hljs-deletion")
         return;
-    if (!versionPattern.test(version.textContent))
```

Comment:
![image](https://user-images.githubusercontent.com/1119693/133611603-91c1945c-ecf5-4698-9c1c-e4be1feb0085.png)

The DOM looks different now. The version is no longer a sub-node of the marker. It's the other way around now. The marker is a sub-node of the version, but the version string itself is located after the marker span. Changing this node would affect other parts as well. In this case "new file mode ..." would be changed as well.

Reply:
Btw: Some mails I get have another signature tag "--" added by some mailing lists after the git send signature. Without proper diff parsing it will never work in all scenarios. But that's nothing new.

Reply:
Putting aside the additional signature, we can adjust based on the new DOM to take the version number and move it into the `hljs-comment` span with the `-- ` separator, so it's coloured the same, no? It should not be difficult if it's always at the end of the `code` block, so I don't think it would affect the `new file mode`?

The additional signature is something else. Hmmm :thinking: What does the DOM looks like in that case? Maybe we can process it the same way, and compromise on having just the last signature in the message marked as a comment (if we drop the version pattern check)? Or keep the pattern check and mark as comment just when we find a git version at the end maybe?

Reply:
The DOM for a "two signatures" scenario follows. The screenshot has been taken after the highlighting took place, so the second one is already marked as comment, while the first (the git signature) stays a deletion.

![image](https://user-images.githubusercontent.com/1119693/133759102-1a62b072-f791-4256-9acd-56c704154966.png)

IOW: Testing for a "version pattern" does not work. There might be more text inside the node holding the deletion/comment spans.

Reply:
Just realized that we have a text-node after the "marker span" which could be used for version pattern matching. I will try that.

Still unsure how we could handle the duplicate or multiple signature thing...

Reply:
Just force-pushed, brings back the pattern matching.

Asking myself if we could use "-- " deletion nodes (no more textContent) followed by a text node  as indication that the two nodes could be merged and marked as comment...

Reply:
Ha, before we start highlighting, we have a node with the "moz-txt-sig" class set, which holds another sub-node with the same class holding the second signature.

I will try if we can remove such nodes before we start highlighting and adding them back afterwards.

Reply:
Should be solved with latest push. @Qeole could you have a look again? We would now support multiple signatures as well. We remove the top level signature DOM node from the DOM right before starting the highlighting itself and add it back afterwards with the "hljs-comment" class set.

The affected commit has been reworded to reflect the final changes.

Reply:
Looks good on a first glance, I'll test it sometime during the weekend :+1:. Thanks!